test/codeu/chat/server/storage/StorageTest.java

```diff
@@ -49,15 +45,71 @@ public void setup(){
 		}
 	}
 
+	//Resets fields for next tests
+	private void reset() throws SQLException{
+		storage.messageTable.destroy();
+		storage.conversationTable.destroy();
+		storage = new Storage(database);
+		conversations.clear();
+	}
+
+	@Before
+	public void setup(){
+		database = new Database("test.db");
+		username = "test";
+		username2 = "test2";
+		conversations = new ArrayList<ConversationData>();
+		storage = new Storage(database);
+	}
+
+	@Test
+	public void testAddingData() throws SQLException{
+		addToStorage(15, 3, username);
+		addToStorage(12, 3, username2);
+		List<DBObject<ConversationSchema>> addedConvos = storage.conversationTable.find(ConversationSchema.UNIQUE_ID, username);
+		assertEquals(addedConvos.size(), 15);
+		assertEquals(conversations.size(), 27);
+		reset();
+	}
+
+	@Test
+	public void testLoadingData() throws SQLException{
+		addToStorage(10, 5, username);
+		ArrayList<ConversationData> testConversations = storage.loadConversations(username);
+		for (int i = 0; i < testConversations.size(); i++){
+			assertTrue(testConversations.get(i).isEqual(conversations.get(i)));
+		}
+		reset();
+	}
+
 	@Test
-	public void testLoadingConversations(){
-		ArrayList<ConversationData> testConversations = new ArrayList<ConversationData>();
-		//For whatever reason storage becomes null when entering this test, so everything done in setup is effectively wiped
-		//Any ideas as to why would be helpful
-		//testConversations = storage.loadConversations(username);
-		//for (int i = 0; i < testConversations.size(); i++){
-		//	assertTrue(testConversations.get(i).isEqual(conversations.get(i)));
-		//}
-		assertTrue(true);
+	public void testMultipleUsers() throws SQLException{
+		addToStorage(8, 3, username);
+		addToStorage(4, 7, username2);
+		ArrayList<ConversationData> userConvos = storage.loadConversations(username);
+		userConvos.addAll(storage.loadConversations(username2));
+		for (int i = 0; i < userConvos.size(); i++){
+			assertTrue(userConvos.get(i).isEqual(conversations.get(i)));
+		}
+		reset();
+	}
+
+	@Test
+	public void testChronological() throws SQLException{
+		addToStorage(3, 11, username);
+		ArrayList<ConversationData> basis = storage.loadConversations(username);
```

Comment:
This test lacks clarity. Why do we want to load conversations twice, and shuffle one of them?

Instead I feel loading them once and comparing them against the set of expected values is a good test.